src/Features/Core/Portable/StackTraceExplorer/ParsedStackFrame.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.StackTraceExplorer
+{
+    /// <summary>
+    /// A line of text that was parsed by <see cref="StackTraceAnalyzer" />
+    /// to provide metadata bout the line. 
+    /// </summary>
+    internal class ParsedStackFrame : ParsedFrame
+    {
+        public ParsedStackFrame(string originalText, TextSpan classSpan, TextSpan methodSpan, TextSpan argsSpan)
+            : base(originalText)
+        {
+            Contract.ThrowIfTrue(classSpan.IsEmpty);
+            Contract.ThrowIfTrue(methodSpan.IsEmpty);
+
+            ClassSpan = classSpan;
+            MethodSpan = methodSpan;
+            ArgsSpan = argsSpan;
+        }
+
+        /// <summary>
+        /// The full classname parsed from the line. 
+        /// e.x: [|Microsoft.CodeAnalysis.Editor.CallstackExplorer|].Example(arg1, arg2)
+        /// </summary>
+        public TextSpan ClassSpan { get; }
+
+        /// <summary>
+        /// The method name span
+        /// e.x: Microsoft.CodeAnalysis.Editor.CallstackExplorer.[|Example|](arg1, arg2)
+        /// </summary>
+        public TextSpan MethodSpan { get; }
+
+        /// <summary>
+        /// The span of comma seperated arguments.
+        /// e.x: Microsoft.CodeAnalysis.Editor.CallstackExplorer.Example([|arg1, arg2|])
+        /// </summary>
+        public TextSpan ArgsSpan { get; }
+
+        public virtual async Task<ISymbol?> ResolveSymbolAsync(Solution solution, CancellationToken cancellationToken)
+        {
+            var className = GetClassText(includeTrailingDot: false);
+            var methodName = GetMethodText();
+
+            foreach (var project in solution.Projects)
+            {
+                var service = project.GetLanguageService<IStackTraceExplorerService>();
+                if (service is null)
+                {
+                    continue;
+                }
+
+                var metadataName = service.GetClassMetadataName(className);
+                var memberName = service.GetMethodSymbolName(methodName);
+
+                var compilation = await project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
we could likely avoid an expensive call to get the compilation if we first check if this type name exists at all in this project.  that can be done cheaply using Project.ContainsSymbolWithName first prior to going to compilation.  Note: this expects a normal C#/VB name (so no arity backticks, or generics, just the name portion).

Reply:
How does `Project.ContainsSymbolWithName` not require a compilation?